### ISSUE TITLE
Build hangs when gradle is invoked with buildDependents task

### ISSUE THREAD

**Issue by dvaske** [cdsap/Talaiot#389]
Hi
We're running builds with the talaiot plugin enabled. In some of our build we only run the [`buildDependents`](https://docs.gradle.org/current/userguide/java_plugin.html#lifecycle_tasks) for the projects which have been touched by new commits. However, when we run these tasks the build is hanging and do not terminate. This do not happen if we disable the talaiot plugin.

I did a small poc project where I can reproduce the issue on approx 1 out of 4 builds. The repo is located here: https://github.com/dvaske/talaiot-multiproject-poc

To reproduce the issue, clone the repository and run `./gradlew :sub-a:buildDependent`

**Comment by dvaske** [comment]
And by the way, if it has anything to say we're using Java 17 termurin distribution, but I just checked with Java 11 and I could also reproduce the issue here, though only in 1 out of 10 builds.

**Comment by cdsap** [comment]
thanks for the report, checking

**Comment by cdsap** [comment]
Hi @dvaske 
I tested the repo and I can reproduce it. Extending the logs to `--debug` I can see outputs like:
```
2023-10-27T18:30:33.831-0700 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :sub-b:buildDependents
2023-10-27T18:31:03.783-0700 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Execution worker Thread 3: acquired lock on worker lease
2023-10-27T18:31:03.783-0700 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Execution worker Thread 3: released lock on worker lease
2023-10-27T18:31:13.765-0700 [LIFECYCLE] [org.gradle.cache.internal.DefaultFileLockManager] 
2023-10-27T18:31:13.765-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2023-10-27T18:31:13.766-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2023-10-27T18:31:13.767-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2023-10-27T18:31:13.769-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2023-10-27T18:31:13.769-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2023-10-27T18:31:13.769-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
```
Deadlock. And related to this issue:
https://github.com/gradle/gradle/issues/26825
 

I'm seeing you're using 1.5.3, last month we released 2.0 which is compatible with the configuration cache. The new release drops some metrics and functionality that are not compatible, one of them is the depedencies by task, the culprit of this issue. I tested with 2.0 and I can't reproduce the issue. 
Let me know if works for you updating to 2.0

**Comment by dvaske** [comment]
Cool, thanks for looking into the issue - We'll update to 2.0, but need to update our gradle scripts to 8.4 first (sigh).
